stats.js

```diff
@@ -2,21 +2,87 @@ const AQ = require('./gofish/aquarium.js');
 const connect = require("./connect.js");
 
 async function stats() {
+    let operation_types = await get_operation_types();
 
-    console.log("Retrieving operation type stats");
-
-    await connect();
-    let operation_types = await AQ.OperationType.all();
-    
-    for ( let i=0; i<operation_types.length; i++ ) {
+    for (let i = 0; i < operation_types.length; i++) {
         let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
         let stats = JSON.parse(response.data);
         console.log(`${i}: ${operation_types[i].category}/${operation_types[i].name}`);
-        for(const [key,val] of Object.entries(stats)) {
+        for (const [key, val] of Object.entries(stats)) {
             console.log("\t" + key + ": " + val);
         }
     }
+}
+
+async function most_recent() {
+    let operation_types = await get_operation_types();
+    let stat_map = {};
+
+    for (let i = 0; i < operation_types.length; i++) {
+        let response = await AQ.get("/operation_types/"+operation_types[i].id+"/stats");
+        let stats = JSON.parse(response.data);
+        console.log(`Processing ${operation_types[i].category}/${operation_types[i].name}`);
+        if (stats.last_run) {
+            stat_map[i] = stats;
+        }
+    }
+
+    let sorted_stats = Object.keys(stat_map).map(function(key) {
```

Comment:
If you just add a stats field to operatation_types[i] when you retrieve the stats, you don't need to build this new object before sorting.

Reply:
done!